.sage/go.mod

```diff
@@ -1,7 +1,7 @@
 module sage
 
-go 1.21
+go 1.22
```

Comment:
Shouldn't we keep this at 1.21 and only bump the toolchain? This way we don't force other dependants of this library to bump to 1.22 but start using 1.22 to build it.

Reply:
I guess that could be a good solution! I was thinking of this, as it would produce the same errors as I describe but in reverse for those wanting to stay on 1.21.

Reply:
I made it into a `fix` too.

Reply:
Or should the toolchain be remove altogether to avid these kind of issues? Is that the proper fix?

Reply:
As I understand from the [docs](https://go.dev/doc/toolchain), if we remove the toolchain then the library will default to building with Go 1.22.

Is this not basically what happens already if we were to not go ahead with this PR?

As I understand bumping the toolchain would ensure we are building with 1.22 (but not necessarily enforce 1.22 on users of it). Not sure it will solve potential issues this PR was originally meant to solve though. Maybe give it a try?